sklearn/cluster/tests/test_feature_agglomeration.py

```diff
@@ -0,0 +1,43 @@
+"""
+Tests for sklearn.cluster._feature_agglomeration
+"""
+# Authors: Sergul Aydore 2017
+import numpy as np
+from sklearn.cluster import FeatureAgglomeration
+from sklearn.utils.testing import assert_true
+from sklearn.utils.testing import assert_array_almost_equal
+
+
+def test_feature_agglomeration():
+    n_clusters = 1
+    X = np.array([0, 0, 1]).reshape(1, 3)  # (n_samples, n_features)
+
+    agglo_mean = FeatureAgglomeration(n_clusters=n_clusters,
+                                      pooling_func=np.mean)
+    agglo_median = FeatureAgglomeration(n_clusters=n_clusters,
+                                        pooling_func=np.median)
+    agglo_mean.fit(X)
+    agglo_median.fit(X)
+    assert_true(np.size(np.unique(agglo_mean.labels_)) == n_clusters)
+    assert_true(np.size(np.unique(agglo_median.labels_)) == n_clusters)
+    assert_true(np.size(agglo_mean.labels_) == X.shape[1])
+    assert_true(np.size(agglo_median.labels_) == X.shape[1])
+
+    # Test transform
+    Xt_mean = agglo_mean.transform(X)
+    Xt_median = agglo_median.transform(X)
+    assert_true(Xt_mean.shape[1] == n_clusters)
+    assert_true(Xt_median.shape[1] == n_clusters)
+    assert_true(Xt_mean == np.array([1/3.]))
```

Comment:
why is 1/3. not seen as a pep8 violation by travis?

Reply:
Because the mean of [0,0,1] is 1/3 which is 0.333333.... It seems to have passed all the checks. Do you have another suggestion for this test or what is your concern exactly?

Reply:
@agramfort, spaces around operators is no longer a strict requirement of PEP8. I would suggest, @sergul, that we'd consider it better style to have spaces around arithmetic binary operators except where leaving spaces out helps clarify order of operations (e.g. `a*b + c`).

Reply:
> spaces around operators is no longer a strict requirement of PEP8

can't we still require it? We have all this code that use spaces and I don't
like changing the rules of the game in the middle... it's also likely to create
unnecessary diffs.

Reply:
I think here we should require it. I suppose we could in general, but we will need to help contributors have the same flake8 settings. See #9121.

Reply:
if we put the ignore in setup.cfg and not in travis build then I think it's easy for contributors to run make flake and see the same errors as on travis. I would favor this.

Reply:
So, do you both mean to change it to ``assert_true(Xt_mean == np.array([1 / 3.]))``?